src/components/KTableData/KTableData.vue

```diff
@@ -0,0 +1,714 @@
+  ButtonAppearance,
+  RowLink,
+  TablePaginationAttributes,
+  PageChangeData,
+  PageSizeChangeData,
+  RowBulkAction,
+  TableState,
+  TableStatePayload,
+  SwrvState,
+  SwrvStateData,
+  SortHandlerFunctionParam,
+} from '@/types'
+import { EmptyStateIconVariants } from '@/types'
+import useUniqueId from '@/composables/useUniqueId'
+import type { RouteLocationRaw } from 'vue-router'
+
+const slots = useSlots()
+
+const { useDebounce, useRequest, useSwrvState, clientSideSorter: defaultClientSideSorter } = useUtilities()
+
+const props = defineProps({
+  /**
+   * KTableView props
+   */
+  resizeColumns: {
+    type: Boolean,
+    default: false,
+  },
+  tablePreferences: {
+    type: Object as PropType<TablePreferences>,
+    default: () => ({}),
+  },
+  rowHover: {
+    type: Boolean,
+    default: true,
+  },
+  rowAttrs: {
+    type: Function as PropType<(row: Record<string, any>) => Record<string, string>>,
+    default: () => ({}),
+  },
+  rowLink: {
+    type: Function as PropType<(row: Record<string, any>) => RowLink>,
+    default: () => ({}),
+  },
+  rowBulkActionEnabled: {
+    type: Function as PropType<(row: Record<string, any>) => RowBulkAction>,
+    default: () => true,
+  },
+  cellAttrs: {
+    type: Function,
+    default: () => ({}),
+  },
+  loading: {
+    type: Boolean,
+    default: false,
+  },
+  emptyStateTitle: {
+    type: String,
+    default: 'No Data',
+  },
+  emptyStateMessage: {
+    type: String,
+    default: 'There is no data to display.',
+  },
+  emptyStateActionRoute: {
+    type: [Object, String] as PropType<RouteLocationRaw | string>,
+    default: null,
+  },
+  emptyStateActionMessage: {
+    type: String,
+    default: '',
+  },
+  emptyStateIconVariant: {
+    type: String as PropType<EmptyStateIconVariant>,
+    default: EmptyStateIconVariants.Default,
+  },
+  emptyStateButtonAppearance: {
+    type: String as PropType<ButtonAppearance>,
+    default: null,
+  },
+  error: {
+    type: Boolean,
+    default: false,
+  },
+  errorStateTitle: {
+    type: String,
+    default: 'An error occurred',
+  },
+  errorStateMessage: {
+    type: String,
+    default: 'Data cannot be displayed due to an error.',
+  },
+  errorStateActionRoute: {
+    type: [Object, String] as PropType<RouteLocationRaw | string>,
+    default: null,
+  },
+  errorStateActionMessage: {
+    type: String,
+    default: '',
+  },
+  headers: {
+    type: Array as PropType<TableDataHeader[]>,
+    default: () => [],
+  },
+  maxHeight: {
+    type: String,
+    default: 'none',
+  },
+  hidePagination: {
+    type: Boolean,
+    default: false,
+  },
+  paginationAttributes: {
+    type: Object as PropType<TablePaginationAttributes>,
+    default: () => ({}),
+  },
+  rowExpandable: {
+    type: Function as PropType<(row: Record<string, any>) => boolean>,
+    default: () => false,
+  },
+  hideHeaders: {
+    type: Boolean,
+    default: false,
+  },
+  nested: {
+    type: Boolean,
+    default: false,
+  },
```

Comment:
suggestion: if you define the props with pure type annotations, you can import the shared table props in both the `KTableView` and `KTableData` components and merge the object with any that are needed specifically in this component:

```ts
// Import this interface
interface MyProps {
  msg?: string
  labels?: string[]
}

const { 
  msg = 'hello', 
  labels = ['one', 'two'],
  // This prop only lives in this component
  total = 0,
} = defineProps<MyProps & {
  // Another prop that only lives here
  total: number
}>()
```

This works for Vue 3.5 and above; for 3.4 you'll need to use `withDefaults` [as described here](https://vuejs.org/guide/typescript/composition-api.html#props-default-values). 

If we use this without consideration for 3.4, we'll need to bump the peerDependency range and ensure that `konnect-ui-apps`, `portal` and `konnect-portal`, `kong-admin`, and mesh are all on 3.5

Reply:
You'll need to check the other repositories above before this merges to ensure they're on 3.5+